epitator/structured_incident_annotator.py

```diff
@@ -0,0 +1,287 @@
+                            incident_aggregation = "cumulative"
+                        elif "new" in column_name:
+                            incident_aggregation = "incremental"
+                        incident_count = utils.parse_spelled_number(value.text)
+                        incident_location = row_incident_location
+                        incident_date = row_incident_date
+                        row_incidents.append(AnnoSpan(value.start, value.end, doc, metadata={
+                            'base_type': incident_base_type,
+                            'aggregation': incident_aggregation,
+                            'value': incident_count,
+                            'attributes': filter(lambda x:x, [count_status]),
+                            'location': incident_location.combined_metadata()['geoname'].to_dict() if incident_location else None,
+                            'dateRange': incident_date.combined_metadata()['datetime_range'] if incident_date else None
+                        }))
+                # If a count is marked as incremental any count in the row above
+                # that value is considered cumulative.
+                max_new_cases = -1
+                max_new_deaths = -1
+                for incident_span in row_incidents:
+                    incident = incident_span.metadata
+                    if incident['aggregation'] == "incremental":
+                        if incident['base_type'] == 'caseCount':
+                            if max_new_cases < incident['value']:
+                                max_new_cases = incident['value']
+                        else:
+                            if max_new_deaths < incident['value']:
+                                max_new_deaths = incident['value']
+                for incident_span in row_incidents:
+                    incident = incident_span.metadata
+                    if incident['aggregation'] == None:
+                        if incident['base_type'] == 'caseCount':
+                            if max_new_cases >= 0 and incident['value'] > max_new_cases:
+                                incident['aggregation'] = 'cumulative'
+                        else:
+                            if max_new_deaths >= 0 and incident['value'] > max_new_deaths:
+                                incident['aggregation'] = 'cumulative'
+                for incident_span in row_incidents:
+                    incident = incident_span.metadata
+                    if incident['aggregation'] == 'cumulative':
+                        incident['type'] = "cumulative" + incident['base_type'][0].upper() + incident['base_type'][1:]
+                    else:
+                        incident['type'] = incident['base_type']
+                    del incident['base_type']
+                    del incident['aggregation']
+                incidents.extend(row_incidents)
+        return {'structured_incidents': AnnoTier(incidents)}
```

Comment:
I skimmed this portion because I've been working on this review for a while, but I'd be interested to discuss how we represent incidents within EpiTator, because we haven't ever done that before, right? They've always only lived in JavaScript?

Reply:
I plan to use the same data-structure for the most part. However, there will just be a "count" property instead of one for cases and one for deaths since that info is in the type property now.